go/ql/lib/ext/os.model.yml

```diff
@@ -46,6 +46,14 @@ extensions:
       pack: codeql/go-all
       extensible: sourceModel
     data:
+      - ["os", "", False, "Environ", "", "", "ReturnValue", "environment", "manual"] # TODO: when sources can have access paths, use .ArrayElement
+      - ["os", "", False, "ExpandEnv", "", "", "ReturnValue", "environment", "manual"]
+      - ["os", "", False, "Getenv", "", "", "ReturnValue", "environment", "manual"]
+      - ["os", "", False, "LookupEnv", "", "", "ReturnValue[0]", "environment", "manual"]
       - ["os", "", False, "Open", "", "", "ReturnValue[0]", "file", "manual"]
       - ["os", "", False, "OpenFile", "", "", "ReturnValue[0]", "file", "manual"]
-      - ["os", "", False, "ReadFile", "", "", "ReturnValue[0]", "file", "manual"]
\ No newline at end of file
+      - ["os", "", False, "ReadFile", "", "", "ReturnValue[0]", "file", "manual"]
+      - ["os", "", False, "UserCacheDir", "", "", "ReturnValue[0]", "environment", "manual"]
+      - ["os", "", False, "UserConfigDir", "", "", "ReturnValue[0]", "environment", "manual"]
+      - ["os", "", False, "UserHomeDir", "", "", "ReturnValue[0]", "environment", "manual"]
+      - ["os", "ProcAttr", False, "Env", "", "", "", "environment", "manual"] # TODO: when sources can have access paths, use .ArrayElement
```

Comment:
```suggestion
```
I suggested this model, but on closer inspection I don't think it's right. Before making a new process you can write to this field to set the environment variables for the new process. By default it's nil, which indicates to use the current environment variables. You might write a modified version of the current environment variables to it and then read from it, but we should be tracking that from wherever you get the current environment variables from.